### PR TITLE
API(program_guard) error message enhancement

### PR DESCRIPTION
**Pull request by Aurelius84**
### program_guard API输入类型检查
原始输入类型检查已经很完善，规范了下报错信息接口，优化为`check_type`

### CODE REVIEW

**Review by liym27 — DISMISSED** [review]
LGTM

File: python/paddle/fluid/framework.py [hunk 1]
```diff
@@ -4987,12 +4987,12 @@ def program_guard(main_program, startup_program=None):
              data = fluid.data(name='image', shape=[None, 784, 784], dtype='float32')
     
     """
-    if not isinstance(main_program, Program):
-        raise TypeError("main_program should be Program")
+    from .data_feeder import check_type
+    check_type(main_program, 'main_program', (Program), 'fluid.program_guard')
```
Comment (zhhsplendid) [inline python/paddle/fluid/framework.py]: Why parentheses "()" around Program ?

Comment (Aurelius84) [reply python/paddle/fluid/framework.py]: `isinstance()` support single type or list of type. I keep same style with `()` of type. Here it's ok if remove `()`.

Comment (zhhsplendid) [reply python/paddle/fluid/framework.py]: But `(Program)` is just `Program`, `(Program, )` is a tuple contains only Program. I suggest to remove () or use `(Program, )`. Other things LGTM

Comment (Aurelius84) [reply python/paddle/fluid/framework.py]: Thanks, fixed~~

**Review by liym27 — APPROVED** [review]

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Review by Xreki — APPROVED** [review]
LGTM

**Pull request merged**